File: 26-RemoveDuplicatesFromSortedArray/main.ts
const removeDuplicates = (nums: number[]): number => {
  if (nums.length === 0) return 0;

  let i: number = 0;
  for (let j: number = 1; j < nums.length; j++) {
    if (nums[i] !== nums[j]) {
      i++;
      nums[i] = nums[j];
    }
  }

  return i + 1;
}

console.log(removeDuplicates([1, 1, 2]));
console.log(removeDuplicates([0, 0, 1, 1, 2, 2, 3, 3, 4]));

File: 26-RemoveDuplicatesFromSortedArray/Main.java
class RemoveDuplicatesFromSortedArray {
  public int removeDuplicates(int[] nums) {
    if (nums.length == 0) {
      return 0;
    }

    int i = 0;
    for (int j = 1; j < nums.length; j++) {
      if (nums[j] != nums[i]) {
        i++;
        nums[i] = nums[j];
      }
    }

    return i + 1;
  }

  public static void main(String[] args) {
    RemoveDuplicatesFromSortedArray sol = new RemoveDuplicatesFromSortedArray();
    System.out.println(sol.removeDuplicates(new int[] {1, 1, 2}));
    System.out.println(sol.removeDuplicates(new int[] {0, 0, 1, 1, 2, 2, 3, 3, 4}));
  }
}